packages/react-components/react-utilities/src/compose/slot.ts

```diff
@@ -0,0 +1,172 @@
+import { isValidElement } from 'react';
+import type {
+  AsIntrinsicElement,
+  ComponentState,
+  ExtractSlotProps,
+  SlotComponent,
+  SlotComponentMetadata,
+  SlotPropsRecord,
+  SlotRenderFunction,
+  SlotShorthandValue,
+  UnknownSlotProps,
+} from './types';
+import { SLOT_COMPONENT_METADATA_SYMBOL } from './constants';
+import * as React from 'react';
+
+export type SlotOptions<Props extends UnknownSlotProps> = {
+  elementType:
+    | React.ComponentType<Props>
+    | (Props extends AsIntrinsicElement<infer As> ? As : keyof JSX.IntrinsicElements);
+  defaultProps?: Partial<Props>;
+};
+
+/**
+ * Creates a slot from a slot shorthand or properties (`props.SLOT_NAME` or `props` itself)
+ *
+ * @param options - optional values you can pass to alter the signature of a slot, those values are:
```

Comment:
```suggestion
 * @param value - ???
 * @param options - optional values you can pass to alter the signature of a slot, those values are:
```

Please also describe `value`